src/main/java/cms/gov/madie/measure/services/TestCaseLockService.java

```diff
@@ -113,4 +104,91 @@ public List<String> unlockByUser(String userName) {
     }
     return deleteMessages;
   }
+
+  public boolean lockTestCases(String measureId, List<String> testCaseIds, String userId) {
+    boolean failure = false;
+    List<TestCaseLock> locks = new ArrayList<>();
+    testCaseIds.forEach(
+        testCaseId -> {
+          validateMeasureAndTestCase(measureId, testCaseId);
+          TestCaseLock lock = makeNewLock(measureId, testCaseId, userId);
+          locks.add(lock);
+        });
+    List<TestCaseLock> insertedLocks = new ArrayList<>();
+    for (TestCaseLock lock : locks) {
+      try {
+        testCaseLockRepository.insert(lock);
+        insertedLocks.add(lock);
+      } catch (DuplicateKeyException ex) {
+        Optional<TestCaseLock> existingLock =
+            testCaseLockRepository.findByTestCaseId(lock.getTestCaseId());
+        if (existingLock.isPresent()) {
+          if (!existingLock.get().getLockedBy().equalsIgnoreCase(userId)) {
+            log.error(
+                "lockTestCases: DuplicateKeyException for testCaseId: [{}], lockedBy: [{}], new userId: [{}]",
+                existingLock.get().getTestCaseId(),
+                existingLock.get().getLockedBy(),
+                userId);
+            failure = true;
+            break;
+          }
+          log.info(
+              "lockTestCases: DuplicateKeyException for testCaseId: [{}], lockedBy is the same as userId: ",
+              existingLock.get().getTestCaseId(),
+              userId);
+        }
+      }
+    }
+    if (failure) {
+      insertedLocks.forEach(
+          lock -> {
+            log.info("lockTestCases: revert lock: delete testCaseId: [{}]", lock.getTestCaseId());
+            testCaseLockRepository.deleteByTestCaseId(lock.getTestCaseId());
+          });
+    }
```

Comment:
May need modification depending what use cases we want this method to handle.

Reply:
@jkotanchik-SB maybe we can have a meeting to talk about it?